src/EnergyPlus/IndoorIceRink.cc

```diff
@@ -0,0 +1,2797 @@
+            }
+        }
+        if (!BeginEnvrnFlag && SystemType == IndirectSystem) MyEnvrnFlagIRink(SysNum) = true;
+
+        if (BeginTimeStepFlag && FirstHVACIteration) {
+            {
+                auto const SELECT_CASE_var(SystemType);
+
+                if (SELECT_CASE_var == DirectSystem) {
+                    ZoneNum = DRink(SysNum).ZonePtr;
+                    ZeroSourceSumHATsurf(ZoneNum) = SumHATsurf(ZoneNum); // Set this to figure what partial part of the load the radiant system meets
+                    for (SurfNum = 1; SurfNum < DRink(SysNum).NumOfSurfaces; ++SurfNum) {
+                        if (Surface(DRink(SysNum).SurfacePtrArray(SurfNum)).Class == SurfaceClass_Floor) {
+                            SurfFloor = DRink(SysNum).SurfacePtrArray(SurfNum);
+                        }
+                    }
+                    QRadSysSrcAvg(SurfFloor) = 0.0;      // Initialize this variable to zero (refrigeration system defaults to off)
+                    LastQRadSysSrc(SurfFloor) = 0.0;     // At the start of a time step, reset to zero so average calculation can begin again
+                    LastSysTimeElapsed(SurfFloor) = 0.0; // At the start of a time step, reset to zero so average calculation can begin again
+                    LastTimeStepSys(SurfFloor) = 0.0;    // At the start of a time step, reset to zero so average calculation can begin again
+                } else if (SystemType == IndirectSystem) {
+                    ZoneNum = IRink(SysNum).ZonePtr;
+                    ZeroSourceSumHATsurf(ZoneNum) = SumHATsurf(ZoneNum); // Set this to figure what partial part of the load the radiant system meets
+                    for (SurfNum = 1; SurfNum < IRink(SysNum).NumOfSurfaces; ++SurfNum) {
+                        if (Surface(IRink(SysNum).SurfacePtrArray(SurfNum)).Class == SurfaceClass_Floor) {
+                            SurfFloor = IRink(SysNum).SurfacePtrArray(SurfNum);
+                        }
+                    }
+                    QRadSysSrcAvg(SurfFloor) = 0.0;      // Initialize this variable to zero (refrigeration system defaults to off)
+                    LastQRadSysSrc(SurfFloor) = 0.0;     // At the start of a time step, reset to zero so average calculation can begin again
+                    LastSysTimeElapsed(SurfFloor) = 0.0; // At the start of a time step, reset to zero so average calculation can begin again
+                    LastTimeStepSys(SurfFloor) = 0.0;    // At the start of a time step, reset to zero so average calculation can begin again
+                } else {
+
+                    ShowSevereError("Refrigeration system entered without specification of type: Direct or Indirect?");
+                    ShowContinueError("Occurs in Refrigeration System=" + DRink(SysNum).Name);
+                    ShowFatalError("Preceding condition causes termination.");
+                }
+            }
+        }
+
+        {
+            auto const SELECT_CASE_var(SystemType);
```

Comment:
I know these select case var blocks are present in the code, but there's no reason to include them in new code.  Just use a switch statement if possible, or at least a regular IF block.